test/t8_cmesh_generator/t8_gtest_cmesh_cartestian_product.hxx

```diff
@@ -0,0 +1,223 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2024 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#ifndef T8_GTEST_CMESH_CREATOR_BASE_HXX
+#define T8_GTEST_CMESH_CREATOR_BASE_HXX
+
+#include <t8_cmesh.h>
+#include <tuple>
+#include <functional>
+#include <iterator>
+#include <vector>
+#include <string>
+
+/**
+ * A base class for cmesh examples.
+ * An example must be comparable to another via == and !=. 
+ * 
+ * For pretty debug output a function to translate the parameters of the example to a string should be provided. 
+ * 
+ * The
+ * 
+ */
+class base_example {
```

Comment:
Do we actually need this base class here? I do not see that much functionality.

Reply:
This base class is necessary so the sum of sets can be produced. Since the examples have different types I can not put them directly in one vector. But they all have the same base class. Therefore I can use a pointer of this base class to address all of them via a vector